cli/listener/src/main.rs

```diff
@@ -12,56 +15,44 @@ use solana_sdk::{
     pubkey::Pubkey,
     signature::Keypair,
 };
-use solido_cli_common::snapshot::{
-    Config, OutputMode, SnapshotClient, SnapshotConfig, SnapshotError,
+use solido_cli_common::{
+    error::{AsPrettyError, Error},
+    prometheus::{write_metric, Metric, MetricFamily},
+    snapshot::{Config, OutputMode, SnapshotClient, SnapshotClientConfig},
 };
+use tiny_http::{Header, Request, Response, Server};
+
+const SOLIDO_ID: &str = "solido";
 
 #[derive(Clap, Debug)]
 pub struct Opts {
+    /// Solido's instance address
+    #[clap(long)]
+    solido_address: Pubkey,
+
     /// URL of cluster to connect to (e.g., https://api.devnet.solana.com for solana devnet)
-    // Overwritten by `GeneralOpts` if None.
     #[clap(long, default_value = "http://127.0.0.1:8899")]
     cluster: String,
 
-    /// Whether to output text or json. [default: "text"]
-    // Overwritten by `GeneralOpts` if None.
-    #[clap(long = "output", possible_values = &["text", "json"])]
-    output_mode: Option<OutputMode>,
-
-    /// Unique name for identifying
-    #[clap(long, default_value = "solido")]
-    pool: String,
-
-    /// Poll frequency in seconds, defaults to 5 minutes.
+    /// Poll frequency in seconds.
     #[clap(long, default_value = "300")]
     poll_frequency_seconds: u32,
 
     /// Location of the SQLite DB file.
-    #[clap(long, default_value = "listener.db")]
+    #[clap(long, default_value = "listener.sqlite3")]
     db_path: String,
-}
-
-struct State {
-    pub solido: Lido,
-}
 
-impl State {
-    pub fn new(
-        config: &mut SnapshotConfig,
-        solido_program_id: &Pubkey,
-        solido_address: &Pubkey,
-    ) -> Result<Self, SnapshotError> {
-        let solido = config.client.get_solido(solido_address)?;
-        Ok(State { solido })
-    }
+    /// Listen address and port for the http server that serves a /metrics endpoint.
```

Comment:
```suggestion
    /// Listen address and port for the http server.
```

In this case it’s not just the metrics, the entire point of this daemon is to serve the APY API.